great_expectations_cloud/agent/actions/compatibility.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from typing import Protocol
+
+import great_expectations as gx
+from packaging.version import Version
+from packaging.version import parse as parse_version
+from typing_extensions import override
+
+
+class VersionRunner(Protocol):
+    def run_checkpoint(self) -> None:
+        ...
+
+    def run_check_datasource_config(self) -> None:
+        ...
+
+
+_VERSION_RUNNERS = {}
+
+
+def register_version_runner(version: str, runner: VersionRunner) -> None:
+    _VERSION_RUNNERS[version] = runner
+
+
+class V0Runner(VersionRunner):
+    @override
+    def run_checkpoint(self) -> None:
+        # ***** THIS IS WHERE THE 0.18 SPECIFIC CODE WOULD GO, MOVED FROM THE ACTION *****
+        print("Running run_checkpoint major version 0 (0.18)")
+
+    @override
+    def run_check_datasource_config(self) -> None:
+        # ***** THIS IS WHERE THE 0.18 SPECIFIC CODE WOULD GO, MOVED FROM THE ACTION *****
+        print("Running run_check_datasource_config major version 0 (0.18)")
```

Comment:
Do we expect that the differences here are just schema/model/config changes or also changes to the workflow?

Reply:
Scratch that - changing registration to be action/event/gx_version instead of the guts of the action as in #148.